### PR TITLE
fix(bazel): Remove monolith rule deps from the Java µgen Bazel rules [ggj]

### PR DESCRIPTION
**Pull request by miraleung**
This PR is needed for the preliminary removal of the monolith rules from googleapis' `switched_rules_by_language`.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -49,7 +49,9 @@ jobs:
         run: bazel --batch test $(bazel query "//src/test/..." | grep "Test$") --noshow_progress --test_output=errors
 
       - name: Integration Tests
-        run: bazel --batch test //test/integration:asset //test/integration:credentials //test/integration:iam //test/integration:kms //test/integration:logging //test/integration:pubsub //test/integration:redis //test/integration:library --noshow_progress
+        run: |
+          bazel clean --expunge --async
+          bazel --batch test //test/integration:asset //test/integration:credentials //test/integration:iam //test/integration:kms //test/integration:logging //test/integration:pubsub //test/integration:redis //test/integration:library --noshow_progress
```
Comment (vam-google) [inline .github/workflows/ci.yaml:54]: Why is each integration tests mentioned individually? Why not just `bazel test //...` for all tests? This would greatly simplify CI & workstation consistency between the tests and will not need any extra knowledge to maintain it (basically I have forgotten to add compute test in this list).

If there is a need to launch integration tests as a separate entity (to separate them from unit tests), please consider using [test_suite](https://docs.bazel.build/versions/main/be/general.html#test_suite) instead of mentioning each test individually.

We also need to add compute tests here, either as part of this PR or separately (I'll add it, if you prefer it to be done separately).

File: test/integration/BUILD.bazel [hunk 2]
```diff
@@ -29,20 +31,20 @@ INTEGRATION_TEST_LIBRARIES = [
...
 
 # Keys must match the values in INTEGRATION_TEST_LIBRARIES above.
 API_GAPIC_TARGETS = {
-    "asset": "@com_google_googleapis//google/cloud/asset/v1:asset_java_gapic",
-    "credentials": "@com_google_googleapis//google/iam/credentials/v1:credentials_java_gapic",
+    "asset": ":asset_java_gapic",
+    "credentials": ":credentials_java_gapic",
     "iam": ":iam_java_gapic",  # Googleapis' LRO does not have a Java Gapic.
     "kms": ":kms_java_gapic",  # Local target because mixins are not rolled out yet.
     "pubsub": ":pubsub_java_gapic",
```
Comment (vam-google) [inline test/integration/BUILD.bazel:43]: Why are some tests coming from googleapis/googleapis-discovery and some are "native" to gapic-generator-java? (i.e. what is special about iam, kms and pubsub?)

File: test/integration/goldens/library/com/google/cloud/example/library/v1/stub/LibraryServiceStubSettings.java [hunk 3]
```diff
@@ -408,28 +401,8 @@ protected LibraryServiceStubSettings(Builder settingsBuilder) throws IOException
...
-              .setInitialRetryDelay(Duration.ofMillis(100L))
-              .setRetryDelayMultiplier(1.3)
-              .setMaxRetryDelay(Duration.ofMillis(60000L))
-              .setInitialRpcTimeout(Duration.ofMillis(60000L))
-              .setRpcTimeoutMultiplier(1.0)
-              .setMaxRpcTimeout(Duration.ofMillis(60000L))
-              .setTotalTimeout(Duration.ofMillis(60000L))
-              .build();
-      definitions.put("retry_policy_0_params", settings);
+      settings = RetrySettings.newBuilder().setRpcTimeoutMultiplier(1.0).build();
```
Comment (vam-google) [inline test/integration/goldens/library/com/google/cloud/example/library/v1/stub/LibraryServiceStubSettings.java:404]: Why is this getting removed? Seems like something that would happen when switching from monolith to microgen, or loosing grpc_service_config somewhere

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (miraleung) [reply .github/workflows/ci.yaml]: Because `bazel test //...` runs all the update (i.e. non-test) targets as well. `test_suite` looks like a good option, will explore this separately.

Adding compute tests SGTM, done.

File: test/integration/BUILD.bazel (hunk 2 above)
Comment (miraleung) [reply test/integration/BUILD.bazel:43]: I added some custom changes to IAM, KMS, and PubSub to exercise edge cases that the `googleapis` target does not cover.

File: test/integration/goldens/library/com/google/cloud/example/library/v1/stub/LibraryServiceStubSettings.java (hunk 3 above)
Comment (miraleung) [reply test/integration/goldens/library/com/google/cloud/example/library/v1/stub/LibraryServiceStubSettings.java]: Fixed, thanks for catching that.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (vam-google) [reply .github/workflows/ci.yaml]: bazel test //... will run only what is marked as `Test = true` in bazel [rule](https://docs.bazel.build/versions/4.1.0/skylark/lib/globals.html#parameters-30) definition. 
The integration tests are defined subject for `bazel test` https://github.com/googleapis/gapic-generator-java/blob/master/rules_bazel/java/integration_test.bzl#L87, `_update` targets are not https://github.com/googleapis/gapic-generator-java/blob/master/rules_bazel/java/integration_test.bzl#L164

Comment (vam-google) [reply .github/workflows/ci.yaml]: bazel test //... will run only what is marked as `Test = true` in bazel [rule](https://docs.bazel.build/versions/4.1.0/skylark/lib/globals.html#parameters-30) definition. 
The integration tests are defined subject for `bazel test` https://github.com/googleapis/gapic-generator-java/blob/master/rules_bazel/java/integration_test.bzl#L87, `_update` targets are not https://github.com/googleapis/gapic-generator-java/blob/master/rules_bazel/java/integration_test.bzl#L164

**Review by vam-google — COMMENTED** [review]
LGTM, but please check the `bazel test //...` topic comment.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (miraleung) [reply .github/workflows/ci.yaml]: Done.

**Review by vam-google — APPROVED** [review]

File: test/integration/goldens/library/com/google/cloud/example/library/v1/stub/LibraryServiceStubSettings.java (hunk 3 above)
Comment (miraleung) [reply test/integration/goldens/library/com/google/cloud/example/library/v1/stub/LibraryServiceStubSettings.java]: On testing again - this will now fail because of the monolith rule removal, since it runs all the non-golden tests from the googleapis' targets. Will make this change on my next pass of removing the monolith's rules entirely from this codebase.
